arches/app/functions/primary_descriptors.py

```diff
@@ -1,18 +1,20 @@
+import logging
 import uuid
 from arches.app.functions.base import BaseFunction
 from arches.app.models import models
-from arches.app.models.tile import Tile
 from arches.app.datatypes.datatypes import DataTypeFactory
 from django.utils.translation import ugettext as _
 
+logger = logging.getLogger(__name__)
+
 
 class AbstractPrimaryDescriptorsFunction(BaseFunction):
-    def get_primary_descriptor_from_nodes(self, resource, config):
+    def get_primary_descriptor_from_nodes(self, resource, config, context=None):
         pass
 
 
 class PrimaryDescriptorsFunction(AbstractPrimaryDescriptorsFunction):
-    def get_primary_descriptor_from_nodes(self, resource, config):
+    def get_primary_descriptor_from_nodes(self, resource, config, context=None):
```

Comment:
That's right, but I think we need to keep it in the function definition because it is now possible to override this function with a custom primary descriptor function. When the `get_primary_descriptor_from_nodes` function gets called in resource.py and api.py we pass in the context. This allows for the possibility that someone may want to pass in the context into their own custom function.